src/ControlSystem/Tags.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/Creators/DomainCreator.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/OptionTags.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace control_system::Tags {
+/// The measurement timescales associated with
+/// domain::Tags::FunctionsOfTime.
+struct MeasurementTimescales : db::SimpleTag {
+  using type = std::unordered_map<
+      std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>;
+
+  static constexpr bool pass_metavariables = true;
+
+  template <typename Metavariables>
+  using option_tags =
+      tmpl::list<domain::OptionTags::DomainCreator<Metavariables::volume_dim>>;
+
+  template <typename Metavariables>
+  static type create_from_options(
+      const std::unique_ptr<::DomainCreator<Metavariables::volume_dim>>&
+          domain_creator) noexcept {
+    std::unordered_map<std::string,
+                       std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+        timescales;
+    for (const auto& function_of_time : domain_creator->functions_of_time()) {
```

Comment:
use structured bindings: `const auto& [f_of_t_name, function_of_time]`?

Reply:
I keep finding more reasons to dislike structured bindings.  Most recently: can't read them in gdb.  If you insist I'll change it, but otherwise I'm leaving this as-is.

Reply:
Nope, up to you